### PR TITLE
added new feature of computer vs human 

### PR DESCRIPTION
**Pull request by user23052036**
Hellow There!

I have added the requested new feature of tic-tac-toe game between human and computer.. and made some slight changes added functions to remove problematic input buffer.

Also made the code even more readable.
Please accept my pull request.

### CODE REVIEW

**Comment by user23052036** [comment]
Sir, could you please review my code please

**Comment by user23052036** [comment]
Bhaiyaa, if you could then please review my pull request.. I rechecked in my system and it is working fine.

File: Code.c [hunk 1]
```diff
@@ -34,26 +38,39 @@ int main()
     if (OS=="nt")
         system("color 09");
     rules();
-    printf("\n\nType 1 to start the game:-\nType 2 to view leader board:-\n");
+    printf("\n\nType 1 to start the game against human:-");
+    printf("\nType 2 to start the game against computer:-\n");
+    printf("Type 3 to view leader board:-\n\n\n");
```
Comment (mishal23) [inline Code.c:43]: nit: please bring the new line character on the line 43 for consistency

File: Code.c [hunk 2]
```diff
@@ -34,26 +38,39 @@ int main()
...
+    {
         case 1:
-            do {
+        {
+            do 
+            {
                 p=fopen("score.txt", "a+");
                 printf("\nEnter name of player1: ");
                 scanf("%s",u1);
+                getchar();
```
Comment (mishal23) [inline Code.c:56]: scanf is already done, why do you need getchar?

File: Code.c [hunk 3]
```diff
@@ -79,33 +96,147 @@ int main()
...
+            int *arr = (int*)malloc(length*sizeof(int));
+            do 
+            {
+                p=fopen("score.txt", "a+");
+                printf("\nEnter name of player1: ");
+                scanf("%s",u1);
+                getchar();
+
+                fprintf(p,"\n%s",u1);
+                printf("My name is user23052036(AI-BOT) lets play o_O");
```
Comment (mishal23) [inline Code.c:145]: let's just keep it as AI-BOT

**Review by mishal23 — CHANGES_REQUESTED** [review]

**Comment by user23052036** [comment]
made all the required changes as requested

**Comment by user23052036** [comment]
Video Demo of the working of human vs computer feature below!



https://github.com/user-attachments/assets/ea8a215b-318e-46db-8ff8-8bb9d99e955e

**Comment by user23052036** [comment]
Sir, could you please review and if possible then merge my code

**Comment by user23052036** [comment]
If you get time please review the pull request by this weekend 🙏

**Review by mishal23 — APPROVED** [review]

**Pull request merged**